v2/tools/generator/internal/conversions/property_conversions.go

```diff
@@ -1934,17 +1948,34 @@ func assignObjectsViaIntermediateObject(
 		return nil, nil
 	}
 
-	// If our two types are not adjacent in our conversion graph, this *IS* the conversion you're looking for
-	earlierName := conversionContext.direction.SelectName(destinationName, sourceName)
-	intermediateName, err := conversionContext.FindNextType(earlierName)
-	if err != nil {
-		return nil, errors.Wrapf(
-			err,
-			"looking up next type for %s",
-			astmodel.DebugDescription(destinationEndpoint.Type()))
+	// Require a path from one name to the next, and work out an intermediate step to break down the conversion
+	var intermediateName astmodel.InternalTypeName
+	if conversionContext.PathExists(sourceName, destinationName) {
+		var err error
```

Comment:
Can move this(and below) var outside the `if` statements and to the error checking after the `if-else` blocks?

Reply:
We could, but I prefer keeping things tightly scoped. 

We need `intermediateName` after the **if**, but we don't need either of the potential errors later on. Moving the `err` variable outside implies a need to access the error again, which isn't correct.